fabric-climbing-v1/src/main/java/net/fabricmc/fabric/api/event/block/ClimbingCallback.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.block;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.util.math.BlockPos;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public interface ClimbingCallback {
```

Comment:
I think it would make sense to split this up into a callback determining whether the entity *can* climb, and then another callback for determining what the climbing speed would be.

Reply:
The idea is that if you say the player is climbing, then you get to determine the climb speed.

Reply:
I think the current approach is good, splitting the callbacks would probably result in duplicated logic for API consumers.

Reply:
That's what I'm going to go with.